qemu/tests/blockdev_inc_backup_resize.py

```diff
@@ -0,0 +1,130 @@
+        )
+
+    def do_full_backup(self):
+        extra_options = {"sync": "full",
+                         "persistent": True,
+                         "auto_disable_bitmap": False}
+        backup_utils.blockdev_batch_backup(
+            self.main_vm,
+            self.source_images,
+            self.full_backups,
+            self.bitmaps,
+            **extra_options)
+
+    def prepare_data_disk(self, tag):
+        """
+        Override this function, only make fs and mount it
+        :param tag: image tag
+        """
+        self.format_data_disk(tag)
+
+    def gen_inc_files(self):
+        return list(map(self.generate_data_file, self.src_img_tags))
+
+    def check_bitmaps(self, node_name, bitmap_name):
+        bitmap = block_dirty_bitmap.get_bitmap_by_name(self.main_vm,
+                                                       node_name,
+                                                       bitmap_name)
+        # check if bitmap exists
+        if bitmap is None:
+            self.test.fail('Failed to get bitmap')
+
+        # check if bitmap is persistent
+        if not bitmap['persistent']:
+            self.test.fail('Bitmap should be persistent')
+
+    def check_image_bitmaps_existed(self):
+        # make sure persistent bitmaps always exist after VM shutdown
+        for tag in self.params.objects("source_images"):
+            disk = self.source_disk_define_by_params(self.params, tag)
+            out = disk.info()
+
+            if out:
+                if self.params['check_bitmaps'] not in out:
+                    self.test.fail('Persistent bitmaps should be in image')
+            else:
+                self.test.error('Error when querying image info with qemu-img')
+
+    def check_image_size(self, node_name, block_size):
+        for d in self.main_vm.monitor.cmd("query-block"):
+            if d['inserted']['node-name'] == node_name:
+                node = d['inserted']['image']
+                break
+        else:
+            self.test.error('Error when querying %s with query-block'
+                            % node_name)
+
+        if int(node['virtual-size']) != block_size:
+            self.test.fail('image size %s != %s after block_resize')
+
+    def block_resize_data_disks(self):
+        for ratio in self.params.objects('disk_change_ratio'):
+            for idx, tag in enumerate(self.src_img_tags):
+                image_params = self.params.object_params(tag)
+                block_size = int(self.src_img_sizes[idx] * float(ratio))
+                args = (None, block_size, self.source_images[idx])
+                self.main_vm.monitor.block_resize(*args)
+                self.check_image_size(self.source_images[idx], block_size)
+                self.check_bitmaps(self.source_images[idx], self.bitmaps[idx])
+
+    def do_test(self):
+        self.do_full_backup()
+        self.gen_inc_files()
+        self.main_vm.destroy()
```

Comment:
Suggest to add 'self.check_image_bitmaps_existed()' to follow step5 of the polarion case.

Reply:
I didn't do the check because another case (https://github.com/autotest/tp-qemu/pull/2361) has already covered the test, so we don't make it an assertion any more in this case